### PR TITLE
[SeaTunnel#1181] Add timeout option for each workflow in case some of them cannot finish normally

### LINKED ISSUE #1181: [Bug] [CI] Sometimes workflow job not finished in time
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

We should add timeout option for each workflow in case some of them cannot finish normally, so that we can release resource in time.

### SeaTunnel Version

The latest (2.0.5-SNAPSHOT)

### SeaTunnel Config

```conf
N/A
```

### Running Command

```shell
N/A
```

### Error Exception

```log
N/A
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by asdf2014**
## Purpose of this pull request

Fixes #1181

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -51,4 +52,4 @@ jobs:
       - name: Autobuild
         uses: github/codeql-action/autobuild@v1
       - name: Perform CodeQL Analysis
-        uses: github/codeql-action/analyze@v1
\ No newline at end of file
+        uses: github/codeql-action/analyze@v1
```
Comment (SteNicholas) [inline .github/workflows/codeql.yaml:55]: There is no any change, right?

Comment (SteNicholas) [inline .github/workflows/codeql.yaml:55]: There is no any change, right?

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yaml (hunk 1 above)
Comment (asdf2014) [reply .github/workflows/codeql.yaml:55]: Yes, just add a blank line for the end of the file

**Review by SteNicholas — APPROVED** [review]

**Comment by SteNicholas** [comment]
LGTM.

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**